testcases/OpTestEEH.py

```diff
@@ -82,8 +93,30 @@ def gather_logs(self):
         c.run_command(cmd)
         c.run_command("dmesg")
 
+    def check_eeh_hit(self):
+        c = self.cv_SYSTEM.sys_get_ipmi_console()
+        tries = 10
+        for i in range(1, tries+1):
+            res = c.run_command("dmesg | grep 'EEH: Frozen';echo $?")
+            if int(res[-1]) == 0:
+                return True
+            time.sleep(1)
+        else:
+            return False
+
+    def check_eeh_removed(self):
+        tries = 30
+        c = self.cv_SYSTEM.sys_get_ipmi_console()
+        for i in range(1, tries+1):
+            res = c.run_command("dmesg | grep 'permanently disabled'; echo $?")
```

Comment:
Hi @stewart-ibm  @pridhiviraj ,

I have run this script on habanaro machine and it works fine. And I see the adapters are removed successfully, as shown below.

[console-pexpect]#dmesg | grep 'EEH: Frozen';echo $?
dmesg | grep 'EEH: Frozen';echo $?
[  603.595186] EEH: Frozen PHB#0-PE#1 detected
0
[console-pexpect]#PE 0000:00:00.0 EEH hit success
dmesg  | grep -i 'EEH: Notify device driver to resume'; echo $?
dmesg  | grep -i 'EEH: Notify device driver to resume'; echo $?
1
[console-pexpect]#Waiting for PE 0000:00:00.0 EEH Completion: (1/60)
dmesg  | grep -i 'EEH: Notify device driver to resume'; echo $?
dmesg  | grep -i 'EEH: Notify device driver to resume'; echo $?
1
[console-pexpect]#Waiting for PE 0000:00:00.0 EEH Completion: (2/60)
dmesg  | grep -i 'EEH: Notify device driver to resume'; echo $?
dmesg  | grep -i 'EEH: Notify device driver to resume'; echo $?
1
[console-pexpect]#Waiting for PE 0000:00:00.0 EEH Completion: (3/60)
dmesg  | grep -i 'EEH: Notify device driver to resume'; echo $?
dmesg  | grep -i 'EEH: Notify device driver to resume'; echo $?
1
[console-pexpect]#Waiting for PE 0000:00:00.0 EEH Completion: (4/60)
dmesg  | grep -i 'EEH: Notify device driver to resume'; echo $?
dmesg  | grep -i 'EEH: Notify device driver to resume'; echo $?
1
[console-pexpect]#Waiting for PE 0000:00:00.0 EEH Completion: (5/60)
dmesg  | grep -i 'EEH: Notify device driver to resume'; echo $?
dmesg  | grep -i 'EEH: Notify device driver to resume'; echo $?
1
[console-pexpect]#Waiting for PE 0000:00:00.0 EEH Completion: (6/60)
dmesg  | grep -i 'EEH: Notify device driver to resume'; echo $?
dmesg  | grep -i 'EEH: Notify device driver to resume'; echo $?
1
[console-pexpect]#Waiting for PE 0000:00:00.0 EEH Completion: (7/60)
dmesg  | grep -i 'EEH: Notify device driver to resume'; echo $?
dmesg  | grep -i 'EEH: Notify device driver to resume'; echo $?
1
[console-pexpect]#Waiting for PE 0000:00:00.0 EEH Completion: (8/60)
dmesg  | grep -i 'EEH: Notify device driver to resume'; echo $?
dmesg  | grep -i 'EEH: Notify device driver to resume'; echo $?
1
[console-pexpect]#Waiting for PE 0000:00:00.0 EEH Completion: (9/60)
dmesg  | grep -i 'EEH: Notify device driver to resume'; echo $?
dmesg  | grep -i 'EEH: Notify device driver to resume'; echo $?
[  619.833582] EEH: Notify device driver to resume
0
[console-pexpect]#ls /sys/bus/pci/devices
ls /sys/bus/pci/devices
0000:00:00.0  0001:00:00.0  0001:01:00.0  0001:02:01.0  0001:02:08.0  0001:02:09.0  0001:02:0a.0  0001:02:10.0  0001:02:11.0  0001:08:00.0  0001:09:00.0  0001:0a:00.0  0001:0b:00.0
[console-pexpect]#PE recovered after first EEH error
cat /proc/powerpc/eeh | tail -n 1
cat /proc/powerpc/eeh | tail -n 1
eeh_slot_resets=3
[console-pexpect]#rm -rf /tmp/opal_msglog;touch /sys/firmware/opal/msglog; cp /sys/firmware/opal/msglog /tmp/opal_msglog
rm -rf /tmp/opal_msglog;touch /sys/firmware/opal/msglog; cp /sys/firmware/opal/msglog /tmp/opal_msglog
[console-pexpect]#dmesg -C
dmesg -C
[console-pexpect]#echo 1:0:4:0:0 > /sys/kernel/debug/powerpc/PCI0000/err_injct && lspci -ns 0000:00:00.0; echo $?
echo 1:0:4:0:0 > /sys/kernel/debug/powerpc/PCI0000/err_injct && lspci -ns 0000:00:00.0; echo $?
0000:00:00.0 0604: 1014:03dc
0
[console-pexpect]#dmesg | grep 'EEH: Frozen';echo $?
dmesg | grep 'EEH: Frozen';echo $?
[  625.414305] EEH: Frozen PHB#0-PE#1 detected
0
[console-pexpect]#PE 0000:00:00.0 EEH hit success
dmesg | grep 'permanently disabled'; echo $?
dmesg | grep 'permanently disabled'; echo $?
               last hour and has been permanently disabled.
0
[console-pexpect]#PE removed successfully

Can we please conclude this patch.